### PR TITLE
Update slides

### PR DESCRIPTION
**Pull request by isaacaka**
Adds overview of PyTorch concepts
Includes mention of other frameworks
Adds example derivative of mse loss
Adds diagram of scatter plot with error lines

### CODE REVIEW

**Review by jatkinson1000 — APPROVED** [review]
Thanks @isaacaka it's nice to seed things like datasets and dataloaders in people's minds before starting the exercises so that they have a feel for what they are doing in some context. Happy for you to merge.
